src/libsyntax/parse/parser.rs

```diff
@@ -528,7 +528,11 @@ impl<'a> Parser<'a> {
         if let Some(directory) = directory {
             parser.directory = directory;
         } else if parser.span != syntax_pos::DUMMY_SP {
-            parser.directory.path = sess.codemap().span_to_unmapped_path(parser.span);
+            // FIXME: this conversion from FileName to PathBuf is weird
+            parser.directory.path = sess.codemap()
+                                        .span_to_unmapped_path(parser.span)
+                                        .to_string()
+                                        .into();
```

Comment:
Why do we need `PathBuf` -> `String` -> `PathBuf` here?

Reply:
lol. I have no idea what I did there. That was probably some intermediate step to get stuff to compile.

Reply:
Ah no, I remember why. This is `FileName -> String -> PathBuf`. I should probably change the `path` field to be a `FileName`

Reply:
And fixed